File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_SpriteFonts.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_SpriteFonts
*/
public class t_SpriteFonts : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public t_SpriteFont[]	m_aFont;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_SpriteFonts() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_aFont = new t_SpriteFont[0];
	}
	public bool read(CReadVariable cVariable) {
		try {
			{
				int n = (int) cVariable.getU8();
				if (n > 255) {
					cVariable.error("array size error in m_aFont");
					return false;
				}
				m_aFont = new t_SpriteFont [n];
				for(int i = 0;i < n;++i) {
					m_aFont[i] = new t_SpriteFont();
					if (!m_aFont[i].read(cVariable)) return false;
				}
			}
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		{	
				if (m_aFont.Length > 255) {
					cVariable.error("array size error in m_aFont");
					return false;
				}
			int n = m_aFont.Length;
			cVariable.put((byte) n);
			for(int i = 0;i < n;++i) {
				if (!m_aFont[i].write(cVariable)) return false;

			}
		}
		return true;
	}
};

File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_SpriteFonts.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
from t_SpriteFont import	t_SpriteFont
#/==========================================================================
#*!
#	@brief	t_SpriteFonts
#/
class t_SpriteFonts:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_aFont = []	#t_SpriteFont
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_aFont = []

	def read(self,cVariable):
		n = cVariable.getU8()
		if (n > 255) :
			cVariable.error('array size error in m_aFont')
			return False
		self.m_aFont = [None]*n
		for i in range(n):
			self.m_aFont[i] = t_SpriteFont()
			if (not self.m_aFont[i].read(cVariable)):
				return False
		return True

	def write(self,cVariable):
		if (len(self.m_aFont) > 255) :
			cVariable.error('array size error in m_aFont')
			return False
		n = len(self.m_aFont)
		cVariable.putU8(n)
		for i in range(n):
			if (not self.m_aFont[i].write(cVariable)):
				return False
		return True

